Firestore/core/src/firebase/firestore/remote/buffered_writer.h

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_REMOTE_BUFFERED_WRITER_H
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_REMOTE_BUFFERED_WRITER_H
+
+#include <functional>
+#include <vector>
+
+#include <grpcpp/support/byte_buffer.h>
+
+namespace firebase {
+namespace firestore {
+namespace remote {
+
+// GRPC doesn't allow issuing a write operation before the previous write has
```

Comment:
This should be rephrased (and reformatted) as a documentation comment for the class. As it stands this seems almost like a bunch of implementation notes without a high-level overview of what it does.

Reply:
Took a stab at rephrasing. If that doesn't help, it would be useful for me if you could say which specific questions it doesn't answer.

Reply:
The new version is much better (and sorry I wasn't more concrete about feedback). The specific thing I was after was that the comment be written from the point of view of how the user should use it (or how it fits into the larger structure of the program) rather than describing the implementation constraints that led to its creation.

I like the new version much better, thanks.